cdap-ui/app/directives/dag/my-dag-ctrl.js

```diff
@@ -15,23 +15,131 @@
  */
 
 angular.module(PKG.name + '.commons')
-  .controller('MyDAGController', function MyDAGController(jsPlumb, $scope) {
+  .controller('MyDAGController', function MyDAGController(jsPlumb, $scope, $timeout, MyDAGFactory, GLOBALS) {
+
+    var vm = this;
+
+    var endpoints = [];
+    var sourceSettings = angular.copy(MyDAGFactory.getSettings(false).source);
+    var sinkSettings = angular.copy(MyDAGFactory.getSettings(false).sink);
+
+    var transformSourceSettings = angular.copy(sourceSettings);
```

Comment:
Note - If we could move this to a factory then that would be useful for anybody using the factory.